compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/IrSourceCompilerForInline.kt

```diff
@@ -161,3 +166,6 @@ private tailrec fun IrDeclaration.isInlineOrInsideInline(): Boolean {
     if (parent !is IrDeclaration) return false
     return parent.isInlineOrInsideInline()
 }
+
+// TODO generate better labels; this is unique (includes the object's address), but not very descriptive
+internal fun IrLoop.nonLocalReturnLabel(forBreak: Boolean): String = "$this\$${if (forBreak) "break" else "continue"}"
```

Comment:
Is it possible to use another approach? I'm worry about hash inside `toString` call (in some rare cases there can be collisions). Maybe some kind of counter?

Reply:
...and more debugable solution

Reply:
Apparently `IrLoop` has a `label` field. Not sure what the point is since loops are referenced directly and not by label after the frontend, but I've reused it to give each loop a name that at least includes the name of the enclosing declaration and a unique counter.

Reply:
> Apparently `IrLoop` has a `label` field. Not sure what the point is since loops are referenced directly and not by label after the frontend, but I've reused it to give each loop a name that at least includes the name of the enclosing declaration and a unique counter.

It's preserved there for the sake of transpiler-like backends such as Kotlin/JS. Not quite sure that Kotlin/JS actually uses it, but they specifically requested it.